### DISCUSSION TITLE
Obtaining Test Data Categories after Running Evaluate?

### DISCUSSION THREAD

**Discussion by VikR0001** [explosion/spaCy#9134] [Help: Coding & Implementations] [answered]
I've successfully added my own data to the textcat_goemotions tutorial code and run it. I'm trying to find the categories assigned by the classifier to my test data. I tried:

```
from spacy.tokens import DocBin

nlp = spacy.blank("en")
doc_bin = DocBin().from_disk("/textcat_goemotions/corpus/test.spacy")  

examples = []  
row_num = 0
for doc in doc_bin.get_docs(nlp.vocab):
    labels = [{"label": ent.start_char, "end": ent.end_char, "label": ent.label_} for ent in doc.ents]    
    examples.append(labels)

display(examples)
```

....but this just returned a blank array.

What is the correct way to see what categories have been assigned to my test data?

**Comment by VikR0001** [comment, marked as answer]
I think I found it -- it looks like it's in doc.cats. :). Is there a function that writes out the doc.cats values to disk for each line of text? It's easy to code it myself of course.

**Reply by polm** [reply]
No, there is no function that does that. 

The `evaluate` command is just to check scores of your model on some dataset on the command line, it's not for inspecting output on individual cases. For actual use of a trained model the expectation is you load it in code and apply it to your data as appropriate.

**Reply by VikR0001** [reply]
Very good. My impressions is that, there is classification of the test data. being conducted by the tutorial code - is that correct? If so where can I find the predicted categories for the test data set? I thought they were in "/textcat_goemotions/corpus/test.spacy", but it appears that's just my original true categories. :)

**Reply by VikR0001** [reply]
"For actual use of a trained model the expectation is you load it in code and apply it to your data as appropriate."

Hmmm.... if I'm reading this correctly, the tutorial app just creates the model, and does not use it to assign predicted categories to the test data. Is that correct?

**Reply by polm** [reply]
When you train a model, the dev set is evaluated periodically to compute scores, which is also used to choose the "best model". The predictions for the test data are not saved. They usually wouldn't be considered useful, I guess they could be if you wanted to compute a confusion matrix or something?

A model you've trained can be used just like a pretrained model. You can do something like this:

```
import spacy
nlp = spacy.load("training/model-best")
doc = nlp("This is my doc")
print(doc.cats) # here are your predictions
```

I think you are assuming that spaCy classifies documents in a batch to some kind of output file, and while you can easily use it that way, that is not a default usage pattern or something we supply tools for. (It should be possible to implement in like a dozen lines though if that's what you need.)
